compiler/src/dotty/tools/dotc/transform/Mixin.scala

```diff
@@ -169,8 +169,10 @@ class Mixin extends MiniPhase with SymTransformer { thisPhase =>
       case Block(stats, expr) =>
         val (scall, inits) = transformConstructor(expr)
         (cpy.Block(tree)(stats, scall), inits)
-      case _ =>
-        val Apply(sel @ Select(New(_), nme.CONSTRUCTOR), args) = tree
+      case Apply(sel @ Select(New(_), nme.CONSTRUCTOR), args) =>
+        val (callArgs, initArgs) = if (tree.symbol.owner.is(Trait)) (Nil, args) else (args, Nil)
+        (superRef(tree.symbol, tree.pos).appliedToArgs(callArgs), initArgs)
+      case Apply(sel @ Select(TypeApply(Select(New(_), aio), List(tpt)), nme.CONSTRUCTOR), args) if aio == defn.Any_asInstanceOf.name =>
```

Comment:
How is this change related to this PR ?

Reply:
Good question. If you remove that case then,
```
testOnly dotty.tools.dotc.CompilationTests -- *tastyBootstrap
```
blows up with a `MatchError` there, the first failing file being `library/src-bootstrapped/scala/Tuple.scala` when compiling `NonEmptyTuple`. There's clearly an interaction with inlining here, but I didn't have time to get to the bottom of it.